spacy/tokens/_retokenize.pyx

```diff
@@ -43,12 +43,12 @@ cdef class Retokenizer:
         attrs = intify_attrs(attrs, strings_map=self.doc.vocab.strings)
         self.merges.append((span, attrs))
 
-    def split(self, Token token, orths, attrs=SimpleFrozenDict()):
+    def split(self, token_index, orths, heads, deps=[], attrs=SimpleFrozenDict()):
```

Comment:
Should we take a `Token` here, instead of the `token_index`? I think taking a `Token` is more consistent with `merge`, which takes a `Span`. I think some code paths can be simpler by taking a `Token` as well. If we take an index, do we end up doing the wrong thing if we make sequential changes, and the token indices end up incorrect?